procbuild/builder.py

```diff
@@ -80,7 +80,172 @@ def checkout(repo, branch, build_path):
+
+    
+    def relocate_build_tools(self):
+        """Move local build tools into temporary directory
+        """
+        self.add_output('Moving proceedings build tools to temp directory...\n')
+        errcode, output = shell('cp -r '
+                                f'{self.master_repo_path}/. {self.build_path}')
+        self.add_output(output)
+        if errcode:
+            self.add_output('[X] Error code %d\n' % errcode)
+            self.build_status = 'Could not move build tools to temp directory'
+            raise BuildError('relocate_build_tools')
+
+    def relocate_static_files(self):
+        for f in self.data_files:
+            shutil.copy(f, self.paper_path)
+    
+    @property
+    def paper_path(self):
+        return joinp(self.build_path, 'papers', self.paper)
+        
+    @property
+    def paper(self):
+        if self.build_path is None:
+            self.add_output('[X] No build path declared: %s\n' % papers)
+            self.build_status = 'No build path declared'
+            raise BuildError('papers')
+        
+        papers = [p for p in iglob(self.build_path + '/papers/*')
+                  if not any(p.endswith(e) for e in excluded)]
+        print(papers)
+        if len(papers)<1:
+            self.add_output('[X] No papers found: %s\n' % papers)
+            self.build_status = 'Paper not found'
+            raise BuildError('papers')
+        
+        # elif len(papers)>1:
+        #     self.add_output('[X] More than one paper found')
+        #     self.build_status = 'More than one paper found'
+        #     raise BuildError('papers')
+        
+        else:
+            return papers[0].split('/')[-1]
+    
+    def run_make_paper_script(self):
+        """Runs the scipy_proceedings make_paper.sh script
+        
+        """
+        self.add_output('[*] Build the paper...\n')
+        errcode, output = shell('./make_paper.sh %s' % self.paper_path, self.build_path)
+        # errcode, output = shell('./make_paper.sh %s' self.paper_path, 
+        #                         self.build_path)
+        self.add_output(output)
+        if errcode:
+            self.add_output('[X] Error code %d\n' % errcode)
+            self.build_status = 'Build failed, make_paper.sh did not succeed'
+            raise BuildError('run_make_paper_script')
+
+    def retrieve_pdf(self):
+        """Collects pdf from temporary directory and moves it to target_path.
+        """
+        output_path = joinp(self.build_path, 'output', self.paper)
+        try:
+            shutil.copy(joinp(output_path, 'paper.pdf'), self.target_path)
+        except IOError:
+            self.add_output('[X] Paper build failed.\n')
+            self.build_status = 'Build failed, no pdf can be found'
+            raise BuildError('retrieve_pdf')
 
+    def build_paper(self):
+        try:
+            with tempfile.TemporaryDirectory() as build_path:
+                self.build_path = build_path
```

Comment:
This section reads very nicely now.  Would it make sense to change some of the methods to private members, e.g. `self._run_make_paper_script`.  My thinking is that we should signal to the user that `build_paper` is really what they want to be calling.

Reply:
that seems reasonable, what I'm hoping is that this will also make it substantially more testable.